### PR TITLE
Fix use after free AV in EventPipe

### PR DESCRIPTION
**Pull request by hoyosjs**
* Disables tracing tests
* Fixes AV

### CODE REVIEW

**Comment by hoyosjs** [comment]
@cshung

**Comment by tommcdon** [comment]
This PR is the Preview 6 fix for #24839

File: src/vm/eventpipeconfiguration.cpp [hunk 1]
```diff
@@ -335,7 +335,8 @@ void EventPipeConfiguration::DeleteSession(EventPipeSession *pSession)
     {
         // Reset the mask of active sessions.
         m_activeSessions &= ~pSession->GetId();
-        delete pSession;
+        // TODO: Re-enable this after fixing the underlying race condition
```
Comment (jorive) [inline src/vm/eventpipeconfiguration.cpp]: :+1:

**Comment by josalem** [comment]
close and open to fix ci

**Comment by jorive** [comment]
Have you tested this with `dotnet-trace`?
- [ ] Single tracing session Enable/Disable 
- [ ] Single tracing session Enable/Disable -> Enable/Disable
- [ ] Enable/Disable multi-sessions?

**Comment by josalem** [comment]
We checked all those scenarios.  The trace output from dotnet-trace is readable and reasonable.  There is a race condition around the hashmap of events when calling from `EventListener`.  The Managed->Native code path leading to `EventPipe::GetNextEvent` does not take locks for a read of that map, and can periodically AV if the map is being modified concurrently.  Simply by keeping the `EventPipeSession` alive is insufficient for this issue.

**Pull request merged**